### PR TITLE
Handle using static

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes 1247

Handle the case where we are passed a type instead of a namespace (using
static).

### CODE REVIEW

**Comment by jmarolf** [comment]
@srivatsn @tmeschter @shyamnamboodiripad @JohnHamby @heejaechang @mavasani

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -388,6 +388,12 @@ protected override string GetDescription(INamespaceOrTypeSymbol namespaceSymbol,
                 return $"using {namespaceString};";
             }
 
+            string staticNamespaceString;
+            if (TryGetStaticNamespaceString(namespaceSymbol, root, false, null, out staticNamespaceString))
+            {
+                return $"using static {staticNamespaceString};";
+            }
+
             // If we get here then neither a namespace or a an extern alias can be added.
```
Comment (VladimirReshetnikov) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: Typo: "a an"

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 2]
```diff
@@ -388,6 +388,12 @@ protected override string GetDescription(INamespaceOrTypeSymbol namespaceSymbol,
...
             // There is no valid string to show to the user and there is 
             // likely a bug in that we should know about.
```
Comment (VladimirReshetnikov) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: "bug in"?

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 3]
```diff
@@ -550,7 +597,14 @@ private static bool ShouldAddExternAlias(ImmutableArray<string> aliases, Compila
 
         private static bool ShouldAddUsing(string usingDirective, CompilationUnitSyntax root)
         {
-            return !root.Usings.Select(u => u.Name.ToString()).Contains(usingDirective);
+            var simpleUsings = root.Usings.Where(u => !u.StaticKeyword.IsKind(SyntaxKind.StaticKeyword));
```
Comment (VladimirReshetnikov) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: `root.Usings.Where(u => u.StaticKeyword.IsKind(SyntaxKind.None))`

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 4]
```diff
@@ -538,9 +561,33 @@ private static bool TryGetNamespaceString(INamespaceOrTypeSymbol namespaceSymbol
                 namespaceString = alias + "::" + namespaceString;
             }
 
+            if (namespaceSymbol is ITypeSymbol)
```
Comment (shyamnamboodiripad) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: [minor] Move this check to the top...

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 5]
```diff
@@ -538,9 +561,33 @@ private static bool TryGetNamespaceString(INamespaceOrTypeSymbol namespaceSymbol
...
+            namespaceString = fullyQualify
+                    ? namespaceSymbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat)
+                    : namespaceSymbol.ToDisplayString();
+
+            if (alias != null)
+            {
+                namespaceString = alias + "::" + namespaceString;
+            }
+
+            if (namespaceSymbol is INamespaceSymbol)
```
Comment (shyamnamboodiripad) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: [minor] Same as above..

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 6]
```diff
@@ -550,7 +597,14 @@ private static bool ShouldAddExternAlias(ImmutableArray<string> aliases, Compila
 
         private static bool ShouldAddUsing(string usingDirective, CompilationUnitSyntax root)
         {
-            return !root.Usings.Select(u => u.Name.ToString()).Contains(usingDirective);
+            var simpleUsings = root.Usings.Where(u => !u.StaticKeyword.IsKind(SyntaxKind.StaticKeyword));
+            return !simpleUsings.Select(u => u.Name.ToString()).Contains(usingDirective);
```
Comment (shyamnamboodiripad) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: return !simpleUsings.Any(u => u.Name.ToString() == usingDirective)?

File: src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs [hunk 7]
```diff
@@ -550,7 +597,14 @@ private static bool ShouldAddExternAlias(ImmutableArray<string> aliases, Compila
...
+        }
+
+        private static bool ShouldAddStaticUsing(string usingDirective, CompilationUnitSyntax root)
+        {
+            var staticUsings = root.Usings.Where(u => u.StaticKeyword.IsKind(SyntaxKind.StaticKeyword));
+            return !staticUsings.Select(u => u.Name.ToString()).Contains(usingDirective);
```
Comment (shyamnamboodiripad) [inline src/Features/CSharp/CodeFixes/AddImport/CSharpAddImportCodeFixProvider.cs]: same as above

**Comment by shyamnamboodiripad** [comment]
:+1:

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests.cs [hunk 8]
```diff
@@ -1322,7 +1322,7 @@ class Test
 
             var expectedText =
 @"using System;
-using Outer;
```
Comment (JohnHamby) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests.cs]: Are the tests sufficient to exercise all the code paths?

**Pull request merged**